ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake

```diff
@@ -63,7 +65,20 @@ function(ament_target_dependencies target)
         # otherwise use the classic CMake variables
         list_append_unique(definitions ${${package_name}_DEFINITIONS})
         list_append_unique(include_dirs ${${package_name}_INCLUDE_DIRS})
-        list(APPEND libraries ${${package_name}_LIBRARIES})
+        if(NOT "${${package_name}_LIBRARY_DIRS}" STREQUAL "")
+          foreach(library ${${package_name}_LIBRARIES})
+            if (NOT IS_ABSOLUTE ${library} OR NOT EXISTS ${library})
+              find_library(lib NAMES ${library} PATHS ${${package_name}_LIBRARY_DIRS} NO_DEFAULT_PATH)
+              if(NOT lib)
+                message(FATAL_ERROR "ament_target_dependencies() ${library} library not found in ${package_name}_LIBRARY_DIRS")
```

Comment:
Shouldn't `${package_name}_LIBRARY_DIRS` be evaluated to show the actual directories searched?

Reply:
I did not want to pollute the output to that extent, but ultimately it's a matter of taste. Variable evaluated in 99e6aab2a8c98de405fb735394b0c11929d182a8